File: Codility/002_01_OddOccurrencesInArray.cpp
// you can use includes, for example:
// #include <algorithm>
#include <unordered_map>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // write your code in C++14 (g++ 6.2.0)
    if(A.size() == 0) return 0;

    int ans = 0;
    int n = A.size();
    unordered_map<int,int> cnt;
    for(int i = 0; i < n; i++) {
        if(cnt.count(A[i])) cnt.erase(A[i]);
        else cnt[A[i]]++;
    }

    ans = cnt.begin()->first;
    return ans;
}

File: Codility/002_01_OddOccurrencesInArray.java
// you can also use imports, for example:
// import java.util.*;
import java.util.HashMap;
import java.util.Set;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 8
        HashMap<Integer, Integer> mark = new HashMap<Integer, Integer>();
        Set<Integer> keys = mark.keySet();

        int ans = 0;
        for(int i = 0; i < A.length; i++) {
            if(mark.containsKey(A[i])) {
                mark.remove(A[i]);
            }
            else {
                mark.put(A[i], 1);
            }
        }

        //print all the keys
        for (Integer key : keys) {
            ans = key;
        }

        return ans;
    }
}
